src/Listeners/ExportMessage.php

```diff
@@ -0,0 +1,153 @@
+     * it implements the ShouldExport interface or not.
+     *
+     * @var bool
+     */
+    public $shouldExport;
+
+    /**
+     * The filesystem disk used to store the message.
+     *
+     * @var string
+     */
+    public $disk;
+
+    /**
+     * The filesystem path used to store the message.
+     *
+     * @var string
+     */
+    public $path;
+
+    /**
+     * The filesystem filename used to store the message.
+     *
+     * @var string
+     */
+    public $filename;
+
+    /**
+     * Create a new listener instance.
+     *
+     * @param Factory $filesystem
+     */
+    public function __construct(Factory $filesystem)
+    {
+        $this->filesystem = $filesystem;
+    }
+
+    /**
+     * Handles the Event when it happens while listening.
+     *
+     * @param MessageSent $event
+     */
+    public function handle(MessageSent $event)
+    {
+        $this->message = $event->message;
+
+        $this->shouldStore = $this->message->_shouldStore ?? false;
+        $this->disk = $this->message->_storageDisk ?? $this->defaultDisk();
+        $this->path = $this->message->_storagePath ?? $this->defaultPath();
+        $this->filename = $this->message->_storageFilename ?? $this->defaultFilename();
+
+        if ($this->shouldStoreMessage()) {
+            $this->storeMessage();
+        }
+    }
+
+    /**
+     * Finds out if wether we should store the mail or not.
+     *
+     * @return bool
+     */
+    protected function shouldStoreMessage(): bool
+    {
+        return $this->shouldStore
+            && config('mail-export.enabled');
+    }
+
+    /**
+     * Actually stores the stringified version of the \Swift_Message including headers,
+     * recipients, subject and body onto the filesystem disk.
+     *
+     * @return void
+     */
+    private function storeMessage()
+    {
+        $this->filesystem
+            ->disk($this->disk)
+            ->put("{$this->path}/{$this->filename}.eml", $this->message->toString());
+
+        event(new MessageStored($this->message, $this->disk, $this->path));
+    }
+
+    /**
+     * Build the default storage disk using the config if none is provided by the developer.
+     *
+     * @return string
+     */
+    private function defaultDisk()
+    {
+        return config('mail-export.disk', config('filesystem.default'));
+    }
+
+    /**
+     * Build the default storage path using the config if none is provided by the developer.
+     *
+     * @return string
+     */
+    private function defaultPath()
+    {
+        return config('mail-export.path');
+    }
+
+    /**
+     * Build some default value for the filename of the message we're about to store
+     * so this can be used if none is provided by the developer from the Mailable.
+     *
+     * @return string
+     */
+    private function defaultFilename(): string
+    {
+        if ($filename = config('mail-export.filename')) {
+            return $filename;
```

Comment:
I think we still need to have a unique element to this, as if someone doesn't set the path to be unique per mailable then multiple mailables with the same filename (which they will be) will just result in them constantly being overwritten.

Reply:
Yeah agreed, this has now been removed.